### ISSUE TITLE
App Update Fails with a generic error message

### ISSUE THREAD

**Issue by ds62987** [ms-iot/iot-core-azure-dm-client#298]
I have an Appx that I've installed on the Raspberry Pi.  This Appx includes the IOTDMClientLib and the necessary calling code as documented.  I have properly installed SystemConfigurator.exe and started it.

Many of the Device management features are working such as timezone settings, listing apps, etc.  The app is also listed as connected and seems to be accepting twin changes fine.

The current version of this Appx that is running is 1.0.17.0.  I am trying to update the app to version 1.0.18.0, but I get this error in the twin: "An error occurred while sending the request".  I have tried everything I can think of but have not been able to successfully update the app.  Any ideas?

More info on my setup:
Raspberry Pi 3 B
Windows 10 Iot Core build 17763
using the latest commit to the Master branch of the repo (as of 12/06/2018)

Here is the current Twin I'm using:

{
  "deviceId": "f1c83f47-b56b-49b9-abd2-f9c934430dd6",
  "etag": "AAAAAAAAACM=",
  "deviceEtag": "NjQ3OTcwNzk=",
  "status": "enabled",
  "statusReason": "Created",
  "statusUpdateTime": "0001-01-01T00:00:00",
  "connectionState": "Connected",
  "lastActivityTime": "0001-01-01T00:00:00",
  "cloudToDeviceMessageCount": 0,
  "authenticationType": "sas",
  "x509Thumbprint": {
    "primaryThumbprint": null,
    "secondaryThumbprint": null
  },
  "version": 421,
  "properties": {
    "desired": {
      "windows": {
        "deviceInfo": {},
        "externalStorage": {
          "connectionString": "DefaultEndpointsProtocol=https;AccountName=xxxx;AccountKey=xxxxxxxx;EndpointSuffix=core.windows.net"
        },
        "apps": {
          "?": {
            "store": false,
            "nonStore": true
          },
          "SmartTower_Iot_Device-uwp_w33rfah112dmp": {
            "pkgFamilyName": "SmartTower.Iot.Device-uwp_w33rfah112dmp",
            "version": "1.0.18.0",
            "startUp": "background",
            "appxSource": "app\\v1.0.18.0\\SmartTower.Iot.Device_1.0.18.0_ARM_Debug.appx",
            "depsSources": "app\\v1.0.18.0\\Microsoft.NET.CoreRuntime.2.1.appx;app\\v1.0.18.0\\Microsoft.VCLibs.ARM.Debug.14.00.appx",
            "certSource": "app\\v1.0.18.0\\SmartTower.Iot.Device_1.0.18.0_ARM_Debug.cer",
            "certStore": "./Device/Vendor/MSFT/RootCATrustedCertificates/TrustedPeople"
          }
        }
      },
      "$metadata": {
        "$lastUpdated": "2018-12-09T19:29:57.4182311Z",
        "$lastUpdatedVersion": 35,
        "windows": {
          "$lastUpdated": "2018-12-09T19:29:57.4182311Z",
          "$lastUpdatedVersion": 35
        }
      },
      "$version": 35
    },
    "reported": {
      "windows": {
        "manageAppLifeCycleCmd": {
          "lastChange": {
            "time": "2018-12-09T18:22:58.3869156Z",
            "state": "failed",
            "errSubSystem": "DeviceManagement",
            "errCode": 1722,
            "errContext": "Failure in SystemConfigurator SendRequest RPC"
          }
        },
        "externalStorage": {
          "lastChange": {
            "time": "2018-12-09T19:29:57.7976312Z",
            "state": "completed"
          },
          "connectionString": "DefaultEndpointsProtocol=https;AccountName=xxxxxxx;AccountKey=xxxxxxx;EndpointSuffix=core.windows.net"
        },
        "deviceInfo": {
          "lastChange": {
            "time": "2018-12-09T19:11:03.6310087Z",
            "state": "completed"
          },
          "batteryRuntime": "-1",
          "batteryRemaining": "-1",
          "batteryStatus": "1",
          "osEdition": "123",
          "secureBootState": "0",
          "totalMemory": "1024",
          "totalStorage": "1428",
          "name": "ds03",
          "processorArchitecture": "5",
          "commercializationOperator": "000-88",
          "displayResolution": "1920x1080",
          "radioSwVer": "Not Present",
          "processorType": "0",
          "platform": "IoTUAP",
          "osVer": "10.0.17763.107",
          "fwVer": "10.0.16248.1001",
          "hwVer": "0.1",
          "oem": "Raspberry Pi",
          "type": "Raspberry Pi 2",
          "lang": "en-US",
          "dmVer": "1.3",
          "model": "Raspberry Pi 3",
          "manufacturer": "RASPBERRY PI",
          "id": "{810D60BE-666A-4A83-8F9E-90D2F1147F23}"
        },
        "deviceHealthAttestation": {},
        "windowsUpdatePolicy": {
          "lastChange": {
            "time": "2018-12-09T19:08:51.416029Z",
            "state": "completed"
          },
          "activeHoursStart": 8,
          "activeHoursEnd": 17,
          "allowAutoUpdate": 6,
          "allowUpdateService": 1,
          "branchReadinessLevel": 16,
          "deferFeatureUpdatesPeriod": 0,
          "deferQualityUpdatesPeriod": 0,
          "pauseFeatureUpdates": 0,
          "pauseQualityUpdates": 0,
          "scheduledInstallDay": 0,
          "scheduledInstallTime": 3,
          "ring": "GeneralAvailability",
          "sourcePriority": "unknown"
        },
        "windowsUpdates": {
          "lastChange": {
            "time": "2018-12-09T19:08:55.482424Z",
            "state": "completed"
          },
          "approved": "<error>",
          "installed": "985b3415-e361-44ca-b747-2f6ad4fc986e",
          "failed": "<error>",
          "installable": "<error>",
          "pendingReboot": "<error>",
          "lastScanTime": "2018-12-09T17:59:52Z",
          "deferUpgrade": false
        },
        "wifi": {
          "lastChange": {
            "time": "2018-12-09T19:09:01.7482885Z",
            "state": "completed"
          },
          "Tag": 90,
          "Status": 0,
          "Configuration": {
            "ApplyFromDeviceTwin": "",
            "ReportToDeviceTwin": "yes",
            "Profiles": {}
          }
        },
        "apps": {
          "SmartTower_Iot_Device-uwp_w33rfah112dmp": {
            "pkgFamilyName": "SmartTower.Iot.Device-uwp_w33rfah112dmp",
            "version": "1.0.17.0",
            "startUp": "none",
            "installDate": "",
            "errorCode": -2146233088,
            "errorMessage": "An error occurred while sending the request."
          },
          "IoTOnboardingTask-uwp_1w720vyc4ccym": {
            "pkgFamilyName": "IoTOnboardingTask-uwp_1w720vyc4ccym",
            "version": "1.0.1712.21001",
            "startUp": "none",
            "installDate": ""
          },
          "Microsoft_NET_CoreRuntime_2_1_8wekyb3d8bbwe": {
            "pkgFamilyName": "Microsoft.NET.CoreRuntime.2.1_8wekyb3d8bbwe",
            "version": "2.1.26124.5",
            "startUp": "none",
            "installDate": ""
          },
          "Microsoft_VCLibs_140_00_Debug_8wekyb3d8bbwe": {
            "pkgFamilyName": "Microsoft.VCLibs.140.00.Debug_8wekyb3d8bbwe",
            "version": "14.0.26428.1",
            "startUp": "none",
            "installDate": ""
          },
          "lastChange": {
            "time": "2018-12-09T19:30:06.4654578Z",
            "state": "pending"
          }
        },
        "eventTracingCollectors": {
          "lastChange": {
            "time": "2018-12-09T19:09:22.8824404Z",
            "state": "failed",
            "errSubSystem": "Unknown",
            "errCode": -2147467261,
            "errContext": "Object reference not set to an instance of an object."
          }
        },
        "timeInfo": {
          "lastChange": {
            "time": "2018-12-09T19:09:31.4229243Z",
            "state": "completed"
          },
          "localTime": "2018-12-09T19:09:35.0000837Z",
          "ntpServer": "time.windows.com,0x9",
          "dynamicDaylightTimeDisabled": false,
          "timeZoneKeyName": "Pacific Standard Time",
          "timeZoneBias": 480,
          "timeZoneStandardBias": 0,
          "timeZoneStandardDate": "0000-11-01T02:00:00Z",
          "timeZoneStandardName": "Pacific Standard Time",
          "timeZoneStandardDayOfWeek": 0,
          "timeZoneDaylightBias": -60,
          "timeZoneDaylightDate": "0000-03-02T02:00:00Z",
          "timeZoneDaylightName": "Pacific Daylight Time",
          "timeZoneDaylightDayOfWeek": 0
        },
        "timeService": {
          "lastChange": {
            "time": "2018-12-09T19:09:39.9252114Z",
            "state": "completed"
          },
          "enabled": "yes",
          "startup": "auto",
          "started": "yes",
          "sourcePriority": "unknown"
        },
        "rebootInfo": {
          "lastChange": {
            "time": "2018-12-09T19:09:48.5554838Z",
            "state": "failed",
            "errSubSystem": "DeviceManagement",
            "errCode": 1,
            "errContext": "Failed to read: SyncML\\SyncBody\\Results\\Item\\Data\\"
          }
        },
        "windowsTelemetry": {
          "lastChange": {
            "time": "2018-12-09T19:10:55.1733636Z",
            "state": "completed"
          },
          "level": "basic"
        },
        "certificates": {
          "lastChange": {
            "time": "2018-12-09T19:11:12.0711057Z",
            "state": "completed"
          },
          "rootCATrustedCertificates_Root": {

          },
          "rootCATrustedCertificates_CA": {

          },
          "rootCATrustedCertificates_TrustedPublisher": {
            "<error>": ""
          },
          "rootCATrustedCertificates_TrustedPeople": {
            "<error>": ""
          },
          "certificateStore_CA_System": {

          },
          "certificateStore_Root_System": {
 
          },
          "certificateStore_My_User": {
            "<error>": ""
          },
          "certificateStore_My_System": {
            "<error>": ""
          }
        }
      },
      "$metadata": {
        "$lastUpdated": "2018-12-09T19:30:08.2925643Z",
        "windows": {
          "$lastUpdated": "2018-12-09T19:30:08.2925643Z",
          "manageAppLifeCycleCmd": {
            "$lastUpdated": "2018-12-09T18:23:00.1480491Z",
            "lastChange": {
              "$lastUpdated": "2018-12-09T18:23:00.1480491Z",
              "time": {
                "$lastUpdated": "2018-12-09T18:23:00.1480491Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T18:23:00.1480491Z"
              },
              "errSubSystem": {
                "$lastUpdated": "2018-12-09T18:23:00.1480491Z"
              },
              "errCode": {
                "$lastUpdated": "2018-12-09T18:23:00.1480491Z"
              },
              "errContext": {
                "$lastUpdated": "2018-12-09T18:23:00.1480491Z"
              }
            }
          },
          "externalStorage": {
            "$lastUpdated": "2018-12-09T19:30:04.0503813Z",
            "lastChange": {
              "$lastUpdated": "2018-12-09T19:30:04.0503813Z",
              "time": {
                "$lastUpdated": "2018-12-09T19:30:04.0503813Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T19:30:04.0503813Z"
              }
            },
            "connectionString": {
              "$lastUpdated": "2018-12-09T19:29:59.8107805Z"
            }
          },
          "deviceInfo": {
            "$lastUpdated": "2018-12-09T19:11:20.2907051Z",
            "lastChange": {
              "$lastUpdated": "2018-12-09T19:11:09.6793659Z",
              "time": {
                "$lastUpdated": "2018-12-09T19:11:09.6793659Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T19:11:09.6793659Z"
              }
            },
            "batteryRuntime": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "batteryRemaining": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "batteryStatus": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "osEdition": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "secureBootState": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "totalMemory": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "totalStorage": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "name": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "processorArchitecture": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "commercializationOperator": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "displayResolution": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "radioSwVer": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "processorType": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "platform": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "osVer": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "fwVer": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "hwVer": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "oem": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "type": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "lang": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "dmVer": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "model": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "manufacturer": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "id": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            }
          },
          "deviceHealthAttestation": {
            "$lastUpdated": "2018-12-09T19:11:20.2907051Z",
            "lastChange": {
              "$lastUpdated": "2018-12-09T19:08:50.9451538Z",
              "time": {
                "$lastUpdated": "2018-12-09T19:08:50.9451538Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T19:08:50.9451538Z"
              }
            }
          },
          "windowsUpdatePolicy": {
            "$lastUpdated": "2018-12-09T19:11:20.2907051Z",
            "lastChange": {
              "$lastUpdated": "2018-12-09T19:08:53.5104866Z",
              "time": {
                "$lastUpdated": "2018-12-09T19:08:53.5104866Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T19:08:53.5104866Z"
              }
            },
            "activeHoursStart": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "activeHoursEnd": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "allowAutoUpdate": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "allowUpdateService": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "branchReadinessLevel": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "deferFeatureUpdatesPeriod": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "deferQualityUpdatesPeriod": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "pauseFeatureUpdates": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "pauseQualityUpdates": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "scheduledInstallDay": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "scheduledInstallTime": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "ring": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "sourcePriority": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            }
          },
          "windowsUpdates": {
            "$lastUpdated": "2018-12-09T19:11:20.2907051Z",
            "lastChange": {
              "$lastUpdated": "2018-12-09T19:08:59.361219Z",
              "time": {
                "$lastUpdated": "2018-12-09T19:08:59.361219Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T19:08:59.361219Z"
              }
            },
            "approved": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "installed": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "failed": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "installable": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "pendingReboot": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "lastScanTime": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "deferUpgrade": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            }
          },
          "wifi": {
            "$lastUpdated": "2018-12-09T19:11:20.2907051Z",
            "lastChange": {
              "$lastUpdated": "2018-12-09T19:09:07.8145902Z",
              "time": {
                "$lastUpdated": "2018-12-09T19:09:07.8145902Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T19:09:07.8145902Z"
              }
            },
            "Tag": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "Status": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "Configuration": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z",
              "ApplyFromDeviceTwin": {
                "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
              },
              "ReportToDeviceTwin": {
                "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
              },
              "Profiles": {
                "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
              }
            }
          },
          "apps": {
            "$lastUpdated": "2018-12-09T19:30:08.2925643Z",
            "SmartTower_Iot_Device-uwp_w33rfah112dmp": {
              "$lastUpdated": "2018-12-09T19:16:17.0366508Z",
              "pkgFamilyName": {
                "$lastUpdated": "2018-12-09T19:16:17.0366508Z"
              },
              "version": {
                "$lastUpdated": "2018-12-09T19:16:17.0366508Z"
              },
              "startUp": {
                "$lastUpdated": "2018-12-09T19:16:17.0366508Z"
              },
              "installDate": {
                "$lastUpdated": "2018-12-09T19:16:17.0366508Z"
              },
              "errorCode": {
                "$lastUpdated": "2018-12-09T19:16:08.5172552Z"
              },
              "errorMessage": {
                "$lastUpdated": "2018-12-09T19:16:08.5172552Z"
              }
            },
            "IoTOnboardingTask-uwp_1w720vyc4ccym": {
              "$lastUpdated": "2018-12-09T19:16:10.6458267Z",
              "pkgFamilyName": {
                "$lastUpdated": "2018-12-09T19:16:10.6458267Z"
              },
              "version": {
                "$lastUpdated": "2018-12-09T19:16:10.6458267Z"
              },
              "startUp": {
                "$lastUpdated": "2018-12-09T19:16:10.6458267Z"
              },
              "installDate": {
                "$lastUpdated": "2018-12-09T19:16:10.6458267Z"
              }
            },
            "Microsoft_NET_CoreRuntime_2_1_8wekyb3d8bbwe": {
              "$lastUpdated": "2018-12-09T19:16:12.7866268Z",
              "pkgFamilyName": {
                "$lastUpdated": "2018-12-09T19:16:12.7866268Z"
              },
              "version": {
                "$lastUpdated": "2018-12-09T19:16:12.7866268Z"
              },
              "startUp": {
                "$lastUpdated": "2018-12-09T19:16:12.7866268Z"
              },
              "installDate": {
                "$lastUpdated": "2018-12-09T19:16:12.7866268Z"
              }
            },
            "Microsoft_VCLibs_140_00_Debug_8wekyb3d8bbwe": {
              "$lastUpdated": "2018-12-09T19:16:14.9115686Z",
              "pkgFamilyName": {
                "$lastUpdated": "2018-12-09T19:16:14.9115686Z"
              },
              "version": {
                "$lastUpdated": "2018-12-09T19:16:14.9115686Z"
              },
              "startUp": {
                "$lastUpdated": "2018-12-09T19:16:14.9115686Z"
              },
              "installDate": {
                "$lastUpdated": "2018-12-09T19:16:14.9115686Z"
              }
            },
            "lastChange": {
              "$lastUpdated": "2018-12-09T19:30:08.2925643Z",
              "time": {
                "$lastUpdated": "2018-12-09T19:30:08.2925643Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T19:30:08.2925643Z"
              }
            }
          },
          "eventTracingCollectors": {
            "$lastUpdated": "2018-12-09T19:09:29.02044Z",
            "lastChange": {
              "$lastUpdated": "2018-12-09T19:09:29.02044Z",
              "time": {
                "$lastUpdated": "2018-12-09T19:09:29.02044Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T19:09:29.02044Z"
              },
              "errSubSystem": {
                "$lastUpdated": "2018-12-09T19:09:29.02044Z"
              },
              "errCode": {
                "$lastUpdated": "2018-12-09T19:09:29.02044Z"
              },
              "errContext": {
                "$lastUpdated": "2018-12-09T19:09:29.02044Z"
              }
            }
          },
          "timeInfo": {
            "$lastUpdated": "2018-12-09T19:11:20.2907051Z",
            "lastChange": {
              "$lastUpdated": "2018-12-09T19:09:37.5127589Z",
              "time": {
                "$lastUpdated": "2018-12-09T19:09:37.5127589Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T19:09:37.5127589Z"
              }
            },
            "localTime": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "ntpServer": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "dynamicDaylightTimeDisabled": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "timeZoneKeyName": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "timeZoneBias": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "timeZoneStandardBias": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "timeZoneStandardDate": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "timeZoneStandardName": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "timeZoneStandardDayOfWeek": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "timeZoneDaylightBias": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "timeZoneDaylightDate": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "timeZoneDaylightName": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "timeZoneDaylightDayOfWeek": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            }
          },
          "timeService": {
            "$lastUpdated": "2018-12-09T19:11:20.2907051Z",
            "lastChange": {
              "$lastUpdated": "2018-12-09T19:09:46.1668248Z",
              "time": {
                "$lastUpdated": "2018-12-09T19:09:46.1668248Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T19:09:46.1668248Z"
              }
            },
            "enabled": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "startup": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "started": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            },
            "sourcePriority": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            }
          },
          "rebootInfo": {
            "$lastUpdated": "2018-12-09T19:10:52.7853626Z",
            "lastChange": {
              "$lastUpdated": "2018-12-09T19:10:52.7853626Z",
              "time": {
                "$lastUpdated": "2018-12-09T19:10:52.7853626Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T19:10:52.7853626Z"
              },
              "errSubSystem": {
                "$lastUpdated": "2018-12-09T19:10:52.7853626Z"
              },
              "errCode": {
                "$lastUpdated": "2018-12-09T19:10:52.7853626Z"
              },
              "errContext": {
                "$lastUpdated": "2018-12-09T19:10:52.7853626Z"
              }
            }
          },
          "windowsTelemetry": {
            "$lastUpdated": "2018-12-09T19:11:20.2907051Z",
            "lastChange": {
              "$lastUpdated": "2018-12-09T19:11:01.2387745Z",
              "time": {
                "$lastUpdated": "2018-12-09T19:11:01.2387745Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T19:11:01.2387745Z"
              }
            },
            "level": {
              "$lastUpdated": "2018-12-09T19:11:20.2907051Z"
            }
          },
          "certificates": {
            "$lastUpdated": "2018-12-09T19:11:20.2907051Z",
            "lastChange": {
              "$lastUpdated": "2018-12-09T19:11:18.148411Z",
              "time": {
                "$lastUpdated": "2018-12-09T19:11:18.148411Z"
              },
              "state": {
                "$lastUpdated": "2018-12-09T19:11:18.148411Z"
              }
            },
           
        }
      },
      "$version": 386
    }
  },
  "capabilities": {
    "iotEdge": false
  }
}

**Comment by gmileka** [comment]
Hi ds62987,

Could you please follow the instructions [here](https://github.com/ms-iot/iot-core-azure-dm-client/blob/master/docs/debugging.md) to collect logs?

The error seems to be originating from an underlying component and not from the DM code itself. We need to narrow down what's exactly failing while installing the new version.

Feel free to reach me at gmileka at/ microsoft.com for a quicker response.

thanks,
george

**Comment by ds62987** [comment]
Ah thank you for that.  I've been going through the docs folder in the repo but I didn't see those instructions.

Yesterday I figured out what the problem was through trial and error.  I was using a prefix on the blobs in the storage container that was causing the issue.  I had "v1.0.18.0" in front of the blobs so they would show up as a folder in Storage Explorer.  

I'm not sure why that was a problem, but after I moved the blobs to the container root I am able to update the Appx on the device.

Thank you for your help!
